src/Components/Date/Date.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState } from 'react'
+import 'react-date-range/dist/styles.css'; // main style file
+import 'react-date-range/dist/theme/default.css'; // theme css file
+import { DateRangePicker } from 'react-date-range';
+import IconButton from '@mui/material/IconButton'
+import { format } from 'date-fns'
+import EventAvailableIcon from '@mui/icons-material/EventAvailable';
+import AddIcon from '@mui/icons-material/Add';
+import RemoveIcon from '@mui/icons-material/Remove';
+import { Button, Typography } from '@mui/material';
+import {  useNavigate, useParams } from "react-router-dom";
+import AvilableRooms from '../AvilableRooms/AvilableRooms';
+
+export default function MyDate() {
+  const navigate = useNavigate();
+  const [open, setOpen] = useState(false)
+  const [count, setCount] = useState(0)
+  const{startDate:initialStartDate,endDate:initialEndDate}=useParams();
+  const [dateRange, setDateRange] = useState([{
+    startDate:initialStartDate? new Date(initialStartDate):new Date() ,
+
+    endDate: initialEndDate?new Date(initialEndDate):new Date(),//convert into Date Objects: default current Date
+    key: 'selection',
+  }])
+ 
+  const handleExplor =()=>{
+   const startDate=format(dateRange[0]?.startDate,'YYY-MM-dd')
+   const endDate=format(dateRange[0]?.endDate,'YYY-MM-dd')
+  //  navigate(`/user/room-details/${roomId}?startDate=${startDate}&endDate=${endDate}`, {
+  //   state: { DetailsRoom:response.data, count
+   
+    
+  //  },
+
+  // });
+
+  
+    navigate(`/user/available-rooms?startDate=${startDate}&endDate=${endDate}`,{state:{ranges:dateRange,count:count}})
+
+
+
+    // navigate(`available-rooms/${startDate}/${endDate}`,{state:{ranges:dateRange,count:count}});
+  }
+  const handleChange = (ranges) => {
+    setDateRange([ranges?.selection])
+  }
+  const handleClick = () => {
+    setOpen((prev) => !prev)
+  }
+  const handleIncrement = () => {
+    setCount(count + 1)
+  }
+  const handleDecrement = () => {
+    setCount(count - 1)
+  }
+  return (
+    <>
+    
+    <Typography
+              variant="h6"
+              sx={{
+                color: "#152C5B",
+                fontSize: "16px",
+                fontWeight: 400,
+                fontFamily: "Poppins",
+              }}
+            >
+              Pick a Date
+            </Typography>
+      <div style={{ display: 'flex', alignItems: "center", marginTop: 5, position: "relative" }}>
+        <IconButton aria-label="" onClick={handleClick}>
+          <EventAvailableIcon sx={{ position: "absolute", left: "100%" }} />
+        </IconButton>
+        <input
```

Comment:
I liked that you restricted the user from entering dates in the past, as we are only allowed to choose a date in the present, however, the capacity field required validation. In the current implementation, we are allowed to enter negative numbers. We should restrict that to only one person and above.